File: Assets/Scripts/AirConsole/Messages/SetSlotMessage.cs
using System;

using Newtonsoft.Json.Linq;

namespace CatFight.AirConsole.Messages
{
    [Serializable]
    public sealed class SetSlotMessage : Message
    {
        public override MessageType type => MessageType.SetSlot;

        public int slotId { get; set; }

        public int itemId { get; set; }

        public SetSlotMessage(JToken data)
            : base(data)
        {
            slotId = (int)data["slotId"];
            itemId = (int)data["itemId"];
        }

        public SetSlotMessage()
        {
        }

        public override string ToString()
        {
            return $"SetSlotMessage({slotId}: {itemId})";
        }
    }
}

File: controller/controller/src/models/messages/setslotmessage.ts
import { IMessage, MessageType } from "./message";

export class SetSlotMessage implements IMessage {

    public type: MessageType = MessageType.SetSlot;

    public slotId: number;

    public itemId: number;

    constructor(slotId: number, itemId: number) {

        this.slotId = slotId;
        this.itemId = itemId;
    }
}
